docs/contributing.md

```diff
@@ -1,3 +1,39 @@
+Contributing to Armory
+======================
+Contributions to Armory are welcomed and highly encouraged!  Armory contains a complex suite of tools that both configure the execution
+environment as well as compose a set of objects (from a configuration file) to be executed in said environment.
+
+Primarily, Armory has two main modes of operation:
+  - Native (also known as `--no-docker`) mode - This uses a pre-set python environment to execute the configuration file
+  - Docker - In this mode of operation, armory composes a Docker image, launches the image, and executed the armory configuration
+    within the image.
+    
+For more details, including how to setup your development environment for either mode of operation see: [Setting up Development Environment](#Setting-up-the-Development-Environment)
+
+Armory Development follows the [GitHub Standard Fork & Pull Request Workflow](https://gist.github.com/Chaser324/ce0505fbed06b947d962).  
+
+Armory uses GitHub Actions to test contributions, for more details see [Armory CI](../.github/ci_test.yml).  Generally it will be most 
+useful to setup the [Armory pre-commit hooks](../tools/pre-commit.sh).  For more information see the [Armory Style Guide](../STYLE.md)
+
+## Setting up the Development Environment
+```bash
+pip install -r test-requirements.txt
+```
+### Docker Operation Mode
+
+### Native Operation Mode
+To get setup you will need to create a [virtual environment](https://docs.python.org/3/library/venv.html).  Once created and activated, you will need 
+to install the testing requirements.  Additionally its often useful to use the `-e` flag so that pip points to your local directory for the code
+```bash
+pip install -e .[tests]
```

Comment:
Unclear to me what the `[tests]` part means.

Reply:
the `[tests]` tells pip to install, in addition to `requirements.txt`, `test-requirements.txt`.

Reply:
looks like `pip install -e .[test]` also works